src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -35,6 +35,12 @@ public class ConfigurationManager<T> : BaseConfigurationManager, IConfigurationM
         private const int ConfigurationRetrieverRunning = 1;
         private int _configurationRetrieverState = ConfigurationRetrieverIdle;
 
+        private readonly TimeProvider _timeProvider = TimeProvider.System;
+
+        // If a refresh is requested, then do the refresh as a blocking operation
+        // not on a background thread.
```

Comment:
nit: maybe a bit more background on the rationale behind the change would be helpful for the future & the impact on the refresh logic.